protos/telemetry_server/telemetry_server.proto

```diff
@@ -13,37 +13,48 @@ option java_outer_classname = "TelemetryServerProto";
-    rpc PublishFlightMode(PublishFlightModeRequest) returns(PublishFlightModeResponse) {}
-    // Publish to 'health' updates.
-    rpc PublishHealth(PublishHealthRequest) returns(PublishHealthResponse) {}
+    rpc PublishFlightMode(PublishFlightModeRequest) returns(PublishFlightModeResponse) { option (mavsdk.options.async_type) = SYNC; }
     // Publish to 'status text' updates.
-    rpc PublishStatusText(PublishStatusTextRequest) returns(PublishStatusTextResponse) {}
+    rpc PublishStatusText(PublishStatusTextRequest) returns(PublishStatusTextResponse) { option (mavsdk.options.async_type) = SYNC; }
     // Publish to 'odometry' updates.
-    rpc PublishOdometry(PublishOdometryRequest) returns(PublishOdometryResponse) {}
+    rpc PublishOdometry(PublishOdometryRequest) returns(PublishOdometryResponse) { option (mavsdk.options.async_type) = SYNC; }
     // Publish to 'position velocity' updates.
-    rpc PublishPositionVelocityNed(PublishPositionVelocityNedRequest) returns(PublishPositionVelocityNedResponse) {}
+    rpc PublishPositionVelocityNed(PublishPositionVelocityNedRequest) returns(PublishPositionVelocityNedResponse) { option (mavsdk.options.async_type) = SYNC; }
     // Publish to 'ground truth' updates.
-    rpc PublishGroundTruth(PublishGroundTruthRequest) returns(PublishGroundTruthResponse) {}
+    rpc PublishGroundTruth(PublishGroundTruthRequest) returns(PublishGroundTruthResponse) { option (mavsdk.options.async_type) = SYNC; }
     // Publish to 'IMU' updates (in SI units in NED body frame).
-    rpc PublishImu(PublishImuRequest) returns(PublishImuResponse) {}
+    rpc PublishImu(PublishImuRequest) returns(PublishImuResponse) { option (mavsdk.options.async_type) = SYNC; }
     // Publish to 'Scaled IMU' updates.
-    rpc PublishScaledImu(PublishScaledImuRequest) returns(PublishScaledImuResponse) {}
+    rpc PublishScaledImu(PublishScaledImuRequest) returns(PublishScaledImuResponse) { option (mavsdk.options.async_type) = SYNC; }
     // Publish to 'Raw IMU' updates.
-    rpc PublishRawImu(PublishRawImuRequest) returns(PublishRawImuResponse) {}
-    // Publish to 'HealthAllOk' updates.
-    rpc PublishHealthAllOk(PublishHealthAllOkRequest) returns(PublishHealthAllOkResponse) {}
+    rpc PublishRawImu(PublishRawImuRequest) returns(PublishRawImuResponse) { option (mavsdk.options.async_type) = SYNC; }
     // Publish to 'unix epoch time' updates.
-    rpc PublishUnixEpochTime(PublishUnixEpochTimeRequest) returns(PublishUnixEpochTimeResponse) {}
+    rpc PublishUnixEpochTime(PublishUnixEpochTimeRequest) returns(PublishUnixEpochTimeResponse) { option (mavsdk.options.async_type) = SYNC; }
+    // Subscribe to ARM/DISARM commands
+    rpc SubscribeArmDisarm(SubscribeArmDisarmRequest) returns(stream ArmDisarmResponse) { option (mavsdk.options.async_type) = ASYNC; }
+    // Subscribe to DO_SET_MODE
+    rpc SubscribeDoSetMode(SubscribeDoSetModeRequest) returns(stream DoSetModeResponse) { option (mavsdk.options.async_type) = ASYNC; }
```

Comment:
Oh, I would have expected this to be in an `ActionServer` plugin. Was that not the intention?

Reply:
Alternatively, you could make an autopilot_server plugin and have it contain both.